ground/src/main/java/com/google/android/ground/system/NetworkManager.kt

```diff
@@ -61,11 +61,13 @@ class NetworkManager @Inject constructor(@ApplicationContext private val context
     }
   }
 
-  /** Returns true iff the device has internet connectivity, false otherwise. */
+  /** Returns true if the device has internet connectivity, false otherwise. */
   @RequiresPermission("android.permission.ACCESS_NETWORK_STATE")
   fun isNetworkConnected(): Boolean {
-    val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-    val networkInfo = cm.activeNetworkInfo
-    return networkInfo?.isConnected ?: false
+    val connectivityManager =
+      context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    val activeNetwork = connectivityManager.activeNetwork
+    val networkCapabilities = connectivityManager.getNetworkCapabilities(activeNetwork)
+    return networkCapabilities?.hasCapability(NET_CAPABILITY_INTERNET) == true
```

Comment:
Let's add a log with the result boolean value.

Reply:
Yes. It would be helpful when debugging the app locally to understand how frequently the function is invoked or if it fails what was the last checked value.